packages/babylon/src/index.js

```diff
@@ -91,3 +111,16 @@ function getParserClass(
   }
   return cls;
 }
+
+function hasModuleSyntax(ast) {
+  return ast.program.body.some(
+    child =>
+      (child.type === "ImportDeclaration" &&
+        (!child.importKind || child.importKind === "value")) ||
+      (child.type === "ExportNamedDeclaration" &&
+        (!child.exportKind || child.exportKind === "value")) ||
+      (child.type === "ExportAllDeclaration" &&
+        (!child.exportKind || child.exportKind === "value")) ||
+      child.type === "ExportDefaultDeclaration",
+  );
+}
```

Comment:
👆 Neat!

Reply:
Yeah this is basically to stay in line with https://github.com/babel/babylon/pull/771 since Flowtype just indiscriminately allows mixing ES6 import/export and CommonJS and it seems fine to allow type import/export inside `sourceType:script` files.

Reply:
@loganfsmyth Who made your avatar!

Reply:
Got it from Matt Cummings. His commissions are open again: https://twitter.com/EiffelArt/status/928017289416495105